tests/test_cli/test_utils/test_package_utils.py

```diff
@@ -0,0 +1,207 @@
+
+    assert try_get_item_target_path(
+        path, author_name, item_type_plural, item_name
+    ) == os.path.join(path, author_name, item_type_plural, item_name)
+
+    with mock.patch("os.path.exists", return_value=True):
+        with pytest.raises(
+            click.ClickException,
+            match=re.escape(
+                'Item "some_skill" already exists in target folder "author/skill".'
+            ),
+        ):
+            try_get_item_target_path(path, author_name, item_type_plural, item_name)
+
+
+class TestListAvailablePackages(BaseAEATestCase):
+    """Test list_available_packages method."""
+
+    use_packages_dir: bool = True
+
+    def test_list_available_packages(
+        self,
+    ) -> None:
+        """Test `list_available_packages`"""
+
+        agent_name = "agent"
+        self.create_agents(agent_name)
+
+        package_list = list_available_packages(Path(self.t / agent_name))
+        assert len(package_list) == 1
+
+        (package_id, _), *_ = package_list
+        assert package_id.name == "signing"
+
+        with mock.patch.object(Path, "is_dir", return_value=False):
+            package_list = list_available_packages(Path(self.t / agent_name))
+            assert len(package_list) == 1
+
+            (package_id, package_path), *_ = package_list
+            assert package_id.name == "signing"
+            assert "vendor" not in package_path.parts
+
+
+def test_update_aea_version_range() -> None:
+    """Test `update_aea_version_range`"""
+
+    package_config = AgentConfig("name", "author", aea_version="0.1.0")
+    update_aea_version_range(package_configuration=package_config)
+
+    assert str(get_current_aea_version()) in package_config.aea_version
+
+
+class TestItemPresentWithHash(BaseAEATestCase):
+    """Test is_item_with_hash_present method."""
+
+    use_packages_dir: bool = True
+
+    def test_is_item_with_hash_present(
+        self,
+    ) -> None:
+        """Test `is_item_with_hash_present`"""
+
+        agent_name = "agent"
+        self.create_agents(agent_name)
+
+        agent_config = self.load_agent_config(agent_name)
+        (signing_protocol,) = agent_config.protocols
+
+        assert is_item_with_hash_present(
+            str(self.packages_dir_path), agent_config, signing_protocol.hash
+        )
+
+        assert (
+            is_item_with_hash_present(str(self.packages_dir_path), agent_config, "Qm")
+            is None
+        )
+
+        dummy_config = mock.MagicMock()
+
+        with mock.patch.object(
+            IPFSHashOnly, "hash_directory", return_value="Qm"
+        ), mock.patch(
+            "aea.cli.utils.config.load_item_config", return_value=dummy_config
+        ):
+            assert (
+                cast(
+                    PublicId,
+                    is_item_with_hash_present(
+                        str(self.t / agent_name), agent_config, "Qm"
+                    ),
+                ).name
+                == "signing"
+            )
```

Comment:
why cast to `PublicId` here but not above?

Reply:
Not sure exactly why, but the mypy only complained about this one